pkg/analysis/completion_test.go

```diff
@@ -336,3 +338,77 @@ s.i`, line: 1, char: 3, expected: []string{"items"}},
 		})
 	}
 }
+
+func TestFindObjectExpression(t *testing.T) {
+	f := newFixture(t)
+	tests := []struct {
+		doc                string
+		line, char         uint32
+		expectedNodeTypes  []string
+		expectedParentTree string
+	}{
+		{
+			doc: `foo.bar()`, char: 4, expectedNodeTypes: []string{"attribute"},
+			expectedParentTree: `
+attribute (N): foo.bar
+  identifier (N): foo
+  . (U): .
+  identifier (N): bar`,
+		},
+
+		{
+			doc: `foo.`, char: 4, expectedNodeTypes: []string{"identifier", "."},
+			expectedParentTree: `
+module (N): foo.
+  ERROR (N): foo.
+    identifier (N): foo
+    . (U): .`,
+		},
+
+		{
+			doc: `baz(foo.)`, char: 8, expectedNodeTypes: []string{"identifier", "."},
+			expectedParentTree: `
+argument_list (N): (foo.)
+  ( (U): (
+  identifier (N): foo
+  ERROR (N): .
+    . (U): .
+  ) (U): )`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.doc, func(t *testing.T) {
+			doc := f.MainDoc(tt.doc)
+			pos := protocol.Position{Line: tt.line, Character: tt.char}
+			pt := query.PositionToPoint(pos)
+
+			n, _ := query.NodeAtPosition(doc, pos)
+			assert.Equal(t, tt.expectedParentTree, PrintNodeTree(doc, n.Parent(), ""))
+			nodes, _ := f.a.nodesForCompletion(doc, n, pt) // simulate the flow
+			nodeTypes := []string{}
+			for _, n := range nodes {
+				nodeTypes = append(nodeTypes, n.Type())
+			}
+			assert.ElementsMatch(t, tt.expectedNodeTypes, nodeTypes)
+
+			objNode := f.a.findObjectExpression(nodes, pt)
+			assert.Equal(t, objNode.Type(), "identifier")
+			assert.Equal(t, doc.Content(objNode), "foo")
+		})
+	}
+}
+
+func PrintNodeTree(d document.Document, n *sitter.Node, indent string) string {
```

Comment:
Please rename to `printNodeTree` so as not to expose a test helper.

Reply:
sure